### PR TITLE
chore: affiliate-sunset-modal

### PR DESCRIPTION
**Pull request by Chef-Yogi**
---

## PR-Codex overview
This PR introduces the `AffiliateSunsetModal` component and related functionality to manage the closure of the PancakeSwap Affiliate Program. It also updates translations and hooks to support this new feature.

### Detailed summary
- Added `AffiliateSunsetModal` component to display closure information.
- Implemented `useAffiliateSunsetList` hook to fetch affiliate list status.
- Updated translations in `translations.json` for affiliate program closure messages.
- Integrated modal presentation based on user acknowledgment and affiliate status.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/web/src/hooks/useAffiliateSunsetList.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { useQuery } from '@tanstack/react-query'
+import { useMemo } from 'react'
+import { Address } from 'viem'
+import { useAccount } from 'wagmi'
+
+interface ZksyncAirDropWhiteListData {
```
Comment (thechefpenguin) [inline apps/web/src/hooks/useAffiliateSunsetList.tsx:6]: could rename this

Comment (Chef-Yogi) [reply apps/web/src/hooks/useAffiliateSunsetList.tsx:6]: this one should be removed, thanks

File: apps/web/src/components/Modal/AffiliateSunsetModal.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    return data[key] || p
+  })
+}
+
+export function AffiliateSunsetModal() {
+  const { t } = useTranslation()
+  const isInList = useUserIsInAffiliateListData()
+  const [ack, setACK] = useUserAcknowledgement('affiliate-referral-sunset-v2')
+  const onConfirm = useCallback(() => setACK(true), [setACK])
+  console.log(isInList, 'isInList')
```
Comment (ChefMomota) [inline apps/web/src/components/Modal/AffiliateSunsetModal.tsx]: Remember remove it before merge

Comment (Chef-Yogi) [reply apps/web/src/components/Modal/AffiliateSunsetModal.tsx]: done

File: apps/web/src/hooks/useAffiliateSunsetList.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        await fetchAffiliateList(account!)
+        return true
+      } catch (error) {
+        return false
+      }
+    },
+    enabled: Boolean(account),
+  })
+
+  return useMemo(() => data, [data])
```
Comment (ChefMomota) [inline apps/web/src/hooks/useAffiliateSunsetList.tsx:29]: we should add a default value right? If account no login.

`return useMemo(() => data ?? false, [data])`

File: apps/web/src/components/Modal/AffiliateSunsetModal.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+      onSuccess={onConfirm}
+    />,
+    false,
+    false,
+    'affiliateSunsetModal',
+  )
+
+  useEffect(() => {
+    if (isInList && ack === false) {
+      onOptionsConfirmModalPresent()
```
Comment (thechefpenguin) [inline apps/web/src/components/Modal/AffiliateSunsetModal.tsx:85]: from an optimization point of view, how about moving the logic in useEffect for opening modal to a Callback passed to the hook that runs if data is true? Something like that would save us an effect. 

@Chef-Yogi

File: apps/web/src/hooks/useAffiliateSunsetList.tsx (hunk 3 above)
Comment (Chef-Yogi) [reply apps/web/src/hooks/useAffiliateSunsetList.tsx]: done

File: apps/web/src/components/Modal/AffiliateSunsetModal.tsx (hunk 4 above)
Comment (thechefpenguin) [reply apps/web/src/components/Modal/AffiliateSunsetModal.tsx:85]: I mean in the hook we already have `useUserIsInAffiliateListData` where we have data from the query

**Review by thechefpenguin — APPROVED** [review]

**Pull request merged**